tsfm_public/toolkit/gluonts_data_wrapper.py

```diff
@@ -0,0 +1,245 @@
+    elif np.dtype == "int":
+        return torch.from_numpy(np)
+
+
+def _torch(*nps):
+    return tuple(np_to_torch(x) for x in nps)
+
+
+class StandardScalingGluonTSDataset:
+    """
+    TTM works best on standard scaled data, especially if fewshot
+    finetuning is being performed.
+    We can utilize the entire available context to do that.
+    This is a global sclaing operation done independently on
+    each channel.
+    """
+
+    def __init__(self) -> None:
+        self.mean = []
+        self.std = []
+
+    def fit(self, train_data: Union[TrainingDataset, InputDataset]):
+        """Calculate the statistics on the historical train data.
+
+        Args:
+            train_data (Union[TrainingDataset, InputDataset]): Iterator with
+                each series of shape [num_channels, seq_len] for multivariate
+                and [seq_len] for univariate.
+        """
+        for batch in tqdm(batcher(train_data, batch_size=1)):
+            if batch[0]["target"].ndim == 1:
+                batch[0]["target"] = batch[0]["target"].reshape(1, -1)  # [1, seq_len]
+            self.mean.append(np.mean(impute_series(batch[0]["target"]), axis=1).reshape(-1, 1))
+            std = np.std(impute_series(batch[0]["target"]), axis=1).reshape(-1, 1)
+            for i in range(std.shape[0]):
+                if std[i] == 0:
+                    std[i] = 1
+            self.std.append(std)
+
+    def transform(self, data: Union[TrainingDataset, InputDataset]):
+        """Apply scaler using calculated statistics.
+
+        Args:
+            data (Union[TrainingDataset, InputDataset]): Iterator with
+                each series of shape [num_channels, seq_len] for multivariate
+                and [seq_len] for univariate.
+
+        Returns:
+            Iternator: With each series transformed.
+        """
+        assert len(self.mean) > 0
+        assert len(self.std) > 0
+
+        out = list(data)
+        for i, _ in tqdm(enumerate(out)):
+            out[i]["target"] = (impute_series(out[i]["target"]) - self.mean[i]) / (self.std[i])
+        return iter(out)
+
+    def inverse_transform(self, data: np.ndarray) -> np.ndarray:
+        """Inverse transform, and bring data to original scale.
+
+        Args:
+            data (np.ndarray): Forecast output of shape [batch, seq_len, num_channels]
+
+        Raises:
+            Exception: If NaN is found in the forecast.
+
+        Returns:
+            np.ndarray: Of shape [batch, seq_len, num_channels].
+        """
+        out = np.zeros(data.shape)
+        for i in tqdm(range((data.shape[0]))):
+            out[i, ...] = data[i, ...] * (self.std[i].T) + self.mean[i].T
+            if np.isnan(out[i, ...]).any():
+                raise Exception("NaN found in forecast!")
+        return out
+
+
+class TorchDatasetFromGluonTSTrainingDataset(Dataset):
+    def __init__(
+        self,
+        gluon_dataset: TrainingDataset,
+        seq_len: int,
```

Comment:
Are these choices consistent with GluonTS? Do we want to introduce different conventions in our repository?

Reply:
Is it possible to leverage our existing torch datasets?